plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -184,17 +194,35 @@ public ConnectorPageSource createPageSource(
                 .map(IcebergColumnHandle.class::cast)
                 .collect(toImmutableList());
 
-        Map<Integer, Optional<String>> partitionKeys = split.getPartitionKeys();
+        HdfsContext hdfsContext = new HdfsContext(session);
+        FileIO fileIO = fileIoProvider.createFileIo(hdfsContext, session.getQueryId());
+        TrinoDeleteFilter deleteFilter = new TrinoDeleteFilter(
+                new DummyFileScanTask(split.getPath(), split.getDeletes()),
+                SchemaParser.fromJson(table.getSchemaJson()),
+                icebergColumns,
```

Comment:
This ends up filtering table schema looking for icebergColumns (columns being read by the query).

- for positional deletes, this is useless, since read columns have nothing to do with positions;
- for equality-based deletes, this seems wrong, since equality delete can use a column that is **not** being read

For the `TrinoDeleteFilter` would it work if we pass `tableSchema` as `requestedSchema` to `DeleteFilter` constructor?
would `org.apache.iceberg.data.DeleteFilter#requiredSchema` still be correct?

Reply:
> for equality-based deletes, this seems wrong, since equality delete can use a column that is not being read

This is where it's used. The DeleteFilter uses this list to see what columns were going to be read anyway by the query, and adds any columns needed to apply the filter to the `DeleteFilter.requiredSchema` list.

For positional deletes, this `requiredSchema` also includes the row number column.

Reply:
> if we pass tableSchema as requestedSchema to DeleteFilter

This would result in all queries reading every column of the data file